### DISCUSSION TITLE
Allow Auto Terms to search more content areas

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/publishpress-taxonomies#1492] [Feature Requests]
We've had several requests for Auto Terms to scan more than just the Title and Content area

Some users have requested the ability to search custom fields or WooCommerce product data, for example
<img width="779" alt="Screen Shot 2022-07-17 at 6 00 17 PM" src="https://user-images.githubusercontent.com/3868028/179426433-66e8d925-d3cc-43d0-9d45-dde1563ca645.png">

**Comment by stevejburge** [comment]
One request for searching GeoDirectory fields

**Comment by stevejburge** [comment]
> Hi could you add more "Find term in" options on the General tab:
> 
> So we could have:
> Post Content
> Post Title
> Post Content and Title
> Post Category
> Post Tag
> Post Content Title and Category
> Post Content Title and Tag
> 
> Maybe you could have tick boxes for each of:
> Content
> Title
> Category
> Tag

https://wordpress.org/support/topic/find-term-in-options/

**Comment by stevejburge** [comment]
One request for files names of media attachments

**Comment by ojopaul** [comment]
@stevejburge 
1. The find terms in "Post title, Post content" etc option only allowing user to select a single option at a time, if we're extending these options, don't you think we should allow multi option select?
2. Can we probably make the lists of options we're adding per https://github.com/TaxoPress/TaxoPress/issues/1550?
Based on your comment above, you mentioned adding the below options
> Post Content
> Post Title
> Post Content and Title
> Post Category
> Post Tag
> Post Content Title and Category
> Post Content Title and Tag

Which makes me believe you may be talking about something else because here's how auto link works for the options:
1. Post Content, Post Title: Find all terms present in the post content based on either all the site terms or terms attached to the post
2. So, how will this work with Category and Tags?

**Comment by stevejburge** [comment]
@ojopaul 

1. Yes, to multiple select
2. Maybe the "tag" and "category" is is misleading.

I wonder if there are other areas that can be included such as custom fields, or ACF fields?

**Comment by ojopaul** [comment]
@stevejburge Yes, ACF fields or Custom fields will work but we may have unreasonable long lists in our option especially if we're displaying all custom fields and that's because custom fields are associated on a post-by-post basis rather than a post-type.

So, for us to get custom fields, we need to get it using a post id as parameter and it may not be same for all post types/post IDs.

For example, we can use [get_metadata( $post_id )](https://developer.wordpress.org/reference/functions/get_metadata/) or [get_post_custom_keys( $post_id )](get_post_custom_keys) to retrieve every custom field stored for that post. 

The most efficient way custom fields could be supported is us adding "Custom Fields" as option and if the option is selected, we provide a text field to enter the fields name otherwise, I don't see any other better solution because even if we decided to make a direct query to wp_postmeta to get all keys, that's going to be a very long options especially for active sites and I don't think it's even efficient as a simple visit to wp_postmeta table will show how long that data could be and call that's not limiting based on specific meta key or post ID is not advisable (Probably why WordPress only support meta key query per post ID basis).

In summary, Custom fields is possible but we need to think of how we're implementing this.
